packages/alchemy/src/middleware/gas-manager.ts

```diff
@@ -1,47 +1,101 @@
 import {
   deepHexlify,
+  filterUndefined,
+  isPercentage,
   resolveProperties,
+  type AccountMiddlewareFn,
+  type Hex,
+  type Percentage,
+  type UserOperationFeeOptions,
   type UserOperationRequest,
 } from "@alchemy/aa-core";
+import { fromHex } from "viem";
 import type { AlchemyProvider } from "../provider.js";
 import type { ClientWithAlchemyMethods } from "./client.js";
+import type { RequestGasAndPaymasterAndDataOverrides } from "./types/index.js";
 
 export interface AlchemyGasManagerConfig {
   policyId: string;
 }
 
+export interface AlchemyGasEstimationOptions {
+  estimateGas: boolean;
+  fallbackGasEstimator?: AccountMiddlewareFn;
+  fallbackFeeDataGetter?: AccountMiddlewareFn;
+}
+
 /**
  * This middleware wraps the Alchemy Gas Manager APIs to provide more flexible UserOperation gas sponsorship.
  *
- * If `estimateGas` is true, it will use `alchemy_requestGasAndPaymasterAndData` to get all of the gas estimates + paymaster data
+ * If `delegateGasEstimation` is true, it will use `alchemy_requestGasAndPaymasterAndData` to get all of the gas estimates + paymaster data
  * in one RPC call.
  *
  * Otherwise, it will use `alchemy_requestPaymasterAndData` to get only paymaster data, allowing you
  * to customize the gas and fee estimation middleware.
  *
  * @param provider - the smart account provider to override to use the alchemy gas manager
  * @param config - the alchemy gas manager configuration
- * @param estimateGas - if true, this will use `alchemy_requestGasAndPaymasterAndData` else will use `alchemy_requestPaymasterAndData`
+ * @param delegateGasEstimation - if true, this will use `alchemy_requestGasAndPaymasterAndData` else will use `alchemy_requestPaymasterAndData`
```

Comment:
wait did this change to an object called `gasEstimationOptions`?

Reply:
updated